### PR TITLE
Feature/trajectory for acc calib

### PR DESCRIPTION
**Pull request by nunoguedelha**
This branch integrates changes to facilitate the trajectory generation required for the calibration of the inertial MTB sensors, as some cleanup and small improvements.

**The major changes for inertial sensors calibration are the following:**
- flagged boundary positions (begin/end of each "row " movement) and publish to a YARP port a "use for fitting" flag for each desired position with the respective timestamps. This makes it easier to separate, in an external app, the measurements targeted to be used for fitting ellipsoid parameters, from those captured while reaching a desired starting point for the next trajectory. It's not crucial, it's a nice-to-have in case you want to discard the back and forth movements around the central row. The name of the output port is `/<local>/<partName>/useFurtherPosForFitting:o`. Format:
```
<envelope>   fittingBit
```
> where `<envelope>` is the usual "counter" followed by the "timestamp" of the published data. The timestamp is synchronised with the one from the inertial sensors measurements output port.

- added two additional parameters to the configuration .ini file, tagged as `returnFlagPort` and `fittingFlagPort`. Through those you can specify the name of the reading ports for collecting, in an external app, the former published data (return point flag) and the new "use for fitting" published flag.
- the app will wait for the input plots, mentioned above, to be connected before starting the movements
- A new mode "gridMapping" was created and doesn't trigger the return to the central position, which is now optional. But, in this case, we still keep a return to centre between the 2 perpendicular grids.

**Description of the more general cleanup or small fixes:**
- easy printing of desired positions for debugging purpose
- removed some warnings
- removed `boring_overflow` and `boringModeInitialized` which seamed to be redundant with the code in `getNewDesiredPosition( )` and was causing unwanted trajectory changes
- some other refactoring for reducing the amount of code or make it easier to read. Any comment is welcome.

**Notes:**
if you want to collect the published data ("use for fitting" and "return point" flags), specify the input port names in the configuration .ini file (tagged as `returnFlagPort` and `fittingFlagPort`). Example of configuration .ini file contents:
```
...
returnFlagPort  /read1
fittingFlagPort  /read2
...
```
In that case, the app will wait for "isTheRobotInReturnPoint:o" and "useFurtherPosForFitting:o" ports to be connected to the specified input ports. The connection is performed by an external process.

### CODE REVIEW

File: src/modules/reachRandomJointPositions/src/module.cpp [hunk 1]
```diff
@@ -42,23 +42,26 @@ using namespace yarp::dev;
...
-        for(int jnt=0; jnt < originalPositions.size(); jnt++ )
+        for(int jnt=0; jnt < int(originalPositions.size()); jnt++ )
         {
             std::string part = controlledJoints[jnt].part_name;
             int axis = controlledJoints[jnt].axis_number;
             pos[part]->setRefSpeed(axis,originalRefSpeeds[jnt]);
-            calib[part]->homingSingleJoint(axis);
+            if( !calib[part]->homingSingleJoint(axis) )
+            {
+                std::cerr << "[ERR] reachRandomJointPositionsModule: homing part " << part << ", axis " << axis << " failed!" << std::endl;
```
Comment (traversaro) [inline src/modules/reachRandomJointPositions/src/module.cpp]: Feel free to use `yError()` in yarp modules.

File: src/modules/reachRandomJointPositions/src/module.cpp [hunk 2]
```diff
@@ -238,129 +254,163 @@ bool reachRandomJointPositionsModule::configure(ResourceFinder &rf)
...
+            std::cout  << "List of desired positions: " << std::endl;
+            for(int i=0; i<int(listOfDesiredPositions.size()); i++)
+            {
+                std::cout << listOfDesiredPositions[i].toString() << std::endl;
+            }
+
+            isTheRobotInReturnPoint.open("/"+moduleName+"/"+controlledJoints[0].part_name+"/isTheRobotInReturnPoint:o");
+            useFurtherPosForFitting.open("/"+moduleName+"/"+controlledJoints[0].part_name+"/useFurtherPosForFitting:o");
+            
+            break;
```
Comment (traversaro) [inline src/modules/reachRandomJointPositions/src/module.cpp]: The break is stil working even if it is another scope?

File: src/modules/reachRandomJointPositions/src/module.cpp [hunk 3]
```diff
@@ -238,129 +254,163 @@ bool reachRandomJointPositionsModule::configure(ResourceFinder &rf)
...
+    }
+
+    //Latch the remote timestamp for synchronising local one
+    this->latchTimestampSync();
+
+    //Wait for "isTheRobotInReturnPoint:o" and "useFurtherPosForFitting:o" to be
+    //connected to another port
+    if( !returnFlagPort.empty() )
+    {
+        while(!Network::isConnected(isTheRobotInReturnPoint.getName(), returnFlagPort)) {};
```
Comment (traversaro) [inline src/modules/reachRandomJointPositions/src/module.cpp]: It better to add some sort of the delay to prevent stressing to much the network?

**Comment by traversaro** [comment]
For me it is ok for merge. @fjandrad can you take a look if this does not break your workflow (it should not).

File: src/modules/reachRandomJointPositions/src/module.cpp (hunk 2 above)
Comment (nunoguedelha) [inline src/modules/reachRandomJointPositions/src/module.cpp]: yes, it works outside the scope. I've checked we actually "break" after this line.

File: src/modules/reachRandomJointPositions/src/module.cpp (hunk 1 above)
Comment (nunoguedelha) [inline src/modules/reachRandomJointPositions/src/module.cpp]: As discussed with you, I'll keep it in mind in a further PR.

File: src/modules/reachRandomJointPositions/src/module.cpp (hunk 3 above)
Comment (nunoguedelha) [inline src/modules/reachRandomJointPositions/src/module.cpp]: indeed, will do it before final merge.

**Pull request merged**